nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/util/JdbcCommon.java

```diff
@@ -284,7 +291,7 @@ public static Schema createSchema(final ResultSet rs, String recordName, boolean
                     break;
 
                 case INTEGER:
-                    if (meta.isSigned(i)) {
+                    if (!dbAdapter.getSupportsMetaDataColumnIsSigned() || meta.isSigned(i)) {
```

Comment:
Your spot on. Originally I had this without the `!`, but in my table I had an INT and when I tried to put it into an Avro Long field it threw an exception.